### PR TITLE
fix golint failures - some packages under /pkg/kubelet

### PR DESCRIPTION
**Pull request by Pingan2017**
**What this PR does / why we need it**:
ref #68026 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
cc @dims 
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go [hunk 1]
```diff
@@ -43,10 +43,10 @@ var matchEverythingRules = []registrationv1beta1.RuleWithOperations{{
 	},
 }}
 
-var sideEffectsUnknown registrationv1beta1.SideEffectClass = registrationv1beta1.SideEffectClassUnknown
-var sideEffectsNone registrationv1beta1.SideEffectClass = registrationv1beta1.SideEffectClassNone
-var sideEffectsSome registrationv1beta1.SideEffectClass = registrationv1beta1.SideEffectClassSome
-var sideEffectsNoneOnDryRun registrationv1beta1.SideEffectClass = registrationv1beta1.SideEffectClassNoneOnDryRun
+var sideEffectsUnknown = registrationv1beta1.SideEffectClassUnknown
```
Comment (yue9944882) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go]: why this?

File: pkg/kubelet/images/image_manager.go [hunk 2]
```diff
@@ -112,11 +113,10 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 			msg := fmt.Sprintf("Container image %q already present on machine", container.Image)
 			m.logIt(ref, v1.EventTypeNormal, events.PulledImage, logPrefix, msg, glog.Info)
 			return imageRef, "", nil
-		} else {
-			msg := fmt.Sprintf("Container image %q is not present with pull policy of Never", container.Image)
-			m.logIt(ref, v1.EventTypeWarning, events.ErrImageNeverPullPolicy, logPrefix, msg, glog.Warning)
-			return "", msg, ErrImageNeverPull
 		}
+		msg := fmt.Sprintf("Container image %q is not present with pull policy of Never", container.Image)
```
Comment (yue9944882) [inline pkg/kubelet/images/image_manager.go]: ditto

File: pkg/kubelet/images/helpers.go [hunk 3]
```diff
@@ -46,5 +46,5 @@ func (ts throttledImageService) PullImage(image kubecontainer.ImageSpec, secrets
 	if ts.limiter.TryAccept() {
 		return ts.ImageService.PullImage(image, secrets)
 	}
-	return "", fmt.Errorf("pull QPS exceeded.")
+	return "", fmt.Errorf("pull QPS exceeded")
```
Comment (yue9944882) [inline pkg/kubelet/images/helpers.go]: ditto

File: hack/.golint_failures [hunk 4]
```diff
@@ -211,7 +211,6 @@ pkg/kubelet/dockershim/network/hostport/testing
 pkg/kubelet/dockershim/network/kubenet
 pkg/kubelet/dockershim/network/testing
 pkg/kubelet/events
-pkg/kubelet/images
```
Comment (yue9944882) [inline hack/.golint_failures]: @Pingan2017 thanks for contributing. what's the purpose of this pull plz? these changes are impacting readability of blame msg

Comment (yue9944882) [reply hack/.golint_failures]: @dims it'll be tons of pulls like this flooding the repo. how about put all the changes into one single large pull?

Comment (dims) [reply hack/.golint_failures]: @yue9944882 one large single pull is not desirable. i'd prefer contributors picking off a few packages at a time and filing a PR

Comment (yue9944882) [reply hack/.golint_failures]: >  i'd prefer contributors picking off a few packages at a time and filing a PR

@dims +1. take `pkg/kubelet` as example, it has ~60 more pkgs to go. i guess we might possibly make it within 5~10 pulls? wdyt?

Comment (dims) [reply hack/.golint_failures]: @yue9944882 +1, feel free to suggest groups of packages for the new folks in the original issue.

**Comment by eparis** [comment]
/ok-to-test
/lgtm
/approve
I like this and hope we can keep a clean repo going forward. Though batching is good.

**Comment by yue9944882** [comment]
@Pingan2017 thanks for fixing this. could u put the apiserver changes into another pull plz? also, u could pick some other pkgs under `pkg/kubelet` together in this pull.

**Comment by Pingan2017** [comment]
@eparis  @yue9944882 revert the apiserver change and add some pkgs under pkg/kubelet. PTAL! Thanks!

**Comment by yue9944882** [comment]
@Pingan2017 thx for fixing this! good job!

/lgtm

File: pkg/kubelet/kuberuntime/kuberuntime_gc.go [hunk 5]
```diff
@@ -31,16 +31,16 @@ import (
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
-// containerGC is the manager of garbage collection.
-type containerGC struct {
+// ContainerGC is the manager of garbage collection.
+type ContainerGC struct {
```
Comment (shubheksha) [inline pkg/kubelet/kuberuntime/kuberuntime_gc.go]: Is this type being exported anywhere? I'm curious why golint thinks this is an error if it was working correctly without the capitalization.

Comment (Pingan2017) [reply pkg/kubelet/kuberuntime/kuberuntime_gc.go]: it is privated. golint thinks the exported func return an private is confused.
' func NewContainerGC(client internalapi.RuntimeService, podStateProvider podStateProvider, manager *kubeGenericRuntimeManager)  *ContainerGC '
keep the type private, and also change the func to private. WDYT? @shubheksha

Comment (shubheksha) [reply pkg/kubelet/kuberuntime/kuberuntime_gc.go]: Is the function `NewContainerGC(..)` being called from elsewhere? If not, I think it makes sense to not make this public.

Comment (Pingan2017) [reply pkg/kubelet/kuberuntime/kuberuntime_gc.go]: fixed. PTAL, Thanks! @shubheksha

**Comment by Pingan2017** [comment]
ping @vishh
 for review , Thanks!

**Review by vishh — APPROVED** [review]
/approve

**Comment by Pingan2017** [comment]
ping  @vishh @yue9944882 @shubheksha 
 could someone help  adding a `lgtm` label ? Thanks!

**Pull request merged**